typebinder/src/contexts/exporter.rs

```diff
@@ -170,25 +171,56 @@ impl ExporterContext<'_> {
         generics: &Generics,
         fields: Vec<Field>,
     ) -> Result<Solved<Vec<ExportStatement>>, TsExportError> {
-        let mut imports = Vec::new();
-        let mut constraints = GenericConstraints::default();
-        let members: Vec<TypeMember> = fields
+        #[derive(Default)]
+        struct Accumulator {
+            members: Vec<TypeMember>,
+            extends_clause: Option<InterfaceExtendsClause>,
+            imports: Vec<ImportEntry>,
+            constraints: GenericConstraints,
+        }
+
+        let Accumulator {
+            members,
+            extends_clause,
+            imports,
+            constraints,
+        } = fields
             .into_iter()
-            .filter_map(|field| {
-                if field.attrs.skip_serializing() {
-                    return None;
+            .try_fold(Accumulator::default(), |mut acc, field| {
+                let Accumulator {
+                    members,
+                    extends_clause,
+                    imports,
+                    constraints,
+                } = &mut acc;
+                if !field.attrs.skip_serializing() {
```

Comment:
You could place that if at the start of the closure before the destructuring and do an early-exit